src/AutoRest.CSharp.V3/Generation/Writers/ClientWriter.cs

```diff
@@ -331,6 +342,144 @@ private void WritePagingOperation(CodeWriter writer, Paging pagingMethod, bool a
+            Parameter[] parameters = lroMethod.CreateParameters;
+
+            writer.WriteXmlDocumentationSummary(originalMethod.Description);
+
+            foreach (Parameter parameter in parameters)
+            {
+                writer.WriteXmlDocumentationParameter(parameter.Name, parameter.Description);
+            }
+
+            writer.Append($"public {responseType} {CreateCreateOperationName(lroMethod.Name)}(");
+            foreach (Parameter parameter in parameters)
+            {
+                WriteParameter(writer, parameter);
+            }
+            writer.RemoveTrailingComma();
+            writer.Line($")");
+
+            using (writer.Scope())
+            {
+                WriteParameterNullChecks(writer, parameters);
+
+                writer.Append($"return {typeof(ArmOperationHelpers)}.Create(");
+                writer.Line($"pipeline, clientDiagnostics, originalResponse, {typeof(RequestMethod)}.{originalMethod.Request.HttpMethod.ToRequestMethodName()}, {originalMethod.Diagnostics.ScopeName:L}, {typeof(FinalStateVia)}.{lroMethod.FinalStateVia}, createOriginalHttpMessage,");
+
+                string valueVariable = "value";
+                const string document = "document";
+                ObjectSerialization? serialization = (lroMethod.OriginalResponse.ResponseBody as ObjectResponseBody)?.Serialization;
+                writer.Line($"(r, c) =>");
+                writer.LineRaw("{");
+                switch (serialization)
+                {
+                    case JsonSerialization jsonSerialization:
+                        writer.Append($"using var {document:D} = ");
+                        writer.Line($"{typeof(JsonDocument)}.Parse(r.ContentStream);");
+                        writer.ToDeserializeCall(
+                            jsonSerialization,
+                            w => w.Append($"document.RootElement"),
+                            ref valueVariable
+                        );
+                        writer.Line($"return {typeof(Response)}.FromValue({valueVariable}, r);");
+                        break;
+                    case XmlElementSerialization xmlSerialization:
+                        writer.Line($"var {document:D} = {typeof(XDocument)}.Load(r.ContentStream, LoadOptions.PreserveWhitespace);");
+                        writer.ToDeserializeCall(
+                            xmlSerialization,
+                            w => w.Append($"document"),
+                            ref valueVariable
+                        );
+                        writer.Line($"return {typeof(Response)}.FromValue({valueVariable}, r);");
+                        break;
+                    default:
+                        writer.Line($"return Response.FromValue(r, r);");
+                        break;
+                }
+                writer.LineRaw("},");
+                writer.Line($"async (r, c) =>");
+                writer.LineRaw("{");
```

Comment:
Any reason you didn't use scope here?

Reply:
I originally wrote it using scope. However, you cannot append to the end of a scope. The scope always writes a newline after the `}`. Both of these sets of curly brackets needed characters after them. When I generated it with scope, it was compile-able, but was ugly when Roslyn when through and tried to prettify it.

Reply:
There is a parameter for that https://github.com/Azure/autorest.csharp/blob/feature/v3/src/AutoRest.CSharp.V3/Generation/Writers/CodeWriter.cs#L29

Reply:
Done.